pinot-controller/src/test/java/org/apache/pinot/controller/api/resources/PinotLogicalTableResourceTest.java

```diff
@@ -0,0 +1,308 @@
+
+    // Verify that the logical table is deleted
+    String getLogicalTableUrl = _controllerRequestURLBuilder.forLogicalTableGet(logicalTable.getTableName());
+    try {
+      ControllerTest.sendGetRequest(getLogicalTableUrl, getHeaders());
+      fail("Logical Table GET request should have failed");
+    } catch (IOException e) {
+      assertTrue(e.getMessage().contains("Logical table not found"));
+    }
+  }
+
+  @Test
+  public void testLogicalTableValidationTests() {
+    String addLogicalTableUrl = _controllerRequestURLBuilder.forLogicalTableCreate();
+    String getLogicalTableNamesUrl = _controllerRequestURLBuilder.forLogicalTableNamesGet();
+
+    // Test logical table name with _OFFLINE and _REALTIME is not allowed
+    LogicalTable logicalTable =
+        getLogicalTable("testLogicalTable_OFFLINE", BROKER_TENANT, PHYSICAL_TABLE_NAMES_WITH_TYPE);
+    try {
+      ControllerTest.sendPostRequest(addLogicalTableUrl, logicalTable.toSingleLineJsonString(), getHeaders());
+      fail("Logical Table POST request should have failed");
+    } catch (IOException e) {
+      assertTrue(e.getMessage().contains("Reason: 'tableName' should not end with _OFFLINE or _REALTIME"));
+    }
+
+    // Test logical table name can not be same as existing physical table name
+    logicalTable =
+        getLogicalTable("test_table_1", BROKER_TENANT, PHYSICAL_TABLE_NAMES_WITH_TYPE);
+    try {
+      ControllerTest.sendPostRequest(addLogicalTableUrl, logicalTable.toSingleLineJsonString(), getHeaders());
+      fail("Logical Table POST request should have failed");
+    } catch (IOException e) {
+      assertTrue(e.getMessage().contains("Table name: test_table_1 already exists"));
+    }
+
+    // Test empty physical table names is not allowed
+    logicalTable =
+        getLogicalTable(LOGICAL_TABLE_NAME, BROKER_TENANT, List.of());
+    try {
+      ControllerTest.sendPostRequest(addLogicalTableUrl, logicalTable.toSingleLineJsonString(), getHeaders());
+      fail("Logical Table POST request should have failed");
+    } catch (IOException e) {
+      assertTrue(e.getMessage().contains("'physicalTableNames' should not be null or empty"));
+    }
+
+    // Test cannot create logical table with same name twice
+    logicalTable = getLogicalTable(LOGICAL_TABLE_NAME, BROKER_TENANT, PHYSICAL_TABLE_NAMES_WITH_TYPE);
+    try {
+      ControllerTest.sendPostRequest(addLogicalTableUrl, logicalTable.toSingleLineJsonString(), getHeaders());
+      String tableNames = ControllerTest.sendGetRequest(getLogicalTableNamesUrl, getHeaders());
+      assertEquals(tableNames, "[\"test_logical_table\"]");
+      // create the same logical table again
+      ControllerTest.sendPostRequest(addLogicalTableUrl, logicalTable.toSingleLineJsonString(), getHeaders());
+      fail("Logical Table POST request should have failed");
+    } catch (IOException e) {
+      assertTrue(e.getMessage().contains("Logical table: test_logical_table already exists"));
+    }
+  }
+
+  @Test
```

Comment:
Can you check if the above tests can be parameterized using DataProviders ? Then all the tests can be run with different combinations of db.name. 

Also a test with the logical table in a database is also required. So the combinations I can think are:


| Logical Table | Physical Tables |
+-------------+----------------+
| No DB             | No DB                |
| No DB            | Some in DB        |
| No DB            | All in DB              |
| DB                  | No DB                 |
| DB                  | Some in DB        |
| DB                  | All in DB              |

I am not listing physical tables in different DBs as "Some DB" should take care of that.